src/hotspot/os/linux/os_linux.cpp

```diff
@@ -1889,7 +1889,7 @@ static bool _print_ascii_file(const char* filename, outputStream* st, unsigned*
   ssize_t bytes;
   buf[32] = '\0';
   unsigned lines = 0;
-  while ((bytes = ::read(fd, buf, sizeof(buf)-1)) > 0) {
+  while (!st->was_truncated() && (bytes = ::read(fd, buf, sizeof(buf)-1)) > 0) {
```

Comment:
Okay so we stop printing if `st` is full but where is the notification that this has happened?

Reply:
Caller needs to take care of it.

In my mind, the caller knows what stream class it uses, so it knows if the stream is one that can truncate. So its his responsibility to handle stream::is_truncated. In this case, the caller that uses a truncatable stream is the dcmd framework, and it handles this.

The printing routines themselves could be completely oblivious to it, I only query truncated in a handful of places to save work - if we know the output stream truncated, there is no reason to continue work printing (and potentially, holding a lock while doing it).

Reply:
Okay, so it isn't clear to me that all of these situations are such that the caller is always a DCmd and that it always reports the truncation due to the attach code changes.

Reply:
Okay.